server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -633,6 +635,8 @@ def _domain(self: Self, *, step: "DomainStep") -> StepContext:
         query: "QueryBuilder" = self.QUERY_CLS.from_(
             Table(step.domain, schema=self._db_schema)
         ).select(*selected_cols)
+        if self._source_rows_subset:
+            query = query.limit(self._source_rows_subset)
```

Comment:
Shoudn't we rather have this as an optional parameter of `materialize` ? That way, we'd limit the global query rather than the first step ? SQL engines should be smart enough to optimize the whole thing rather than processing all rows, right ? I'm afraid we might get "the preview returns nothing but I know I have data for the given filters" issues again.

If we get many complaints that the whole thing is too slow, we can still apply the limit in the domain step.

Also, on a join or append step, we'd get LIMIT rows per `domain` step, whereas with a limit on the whole query we'd get LIMIT rows in total